onnxruntime/core/providers/nnapi/nnapi_builtin/builders/node_attr_helper.cc

```diff
@@ -0,0 +1,104 @@
+//
+// Created by daquexian on 8/3/18.
+//
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "core/common/safeint.h"
+#include "node_attr_helper.h"
+
+using std::string;
+using std::vector;
+
+NodeAttrHelper::NodeAttrHelper(const ONNX_NAMESPACE::NodeProto& proto) : node_(proto) {
+}
+
+float NodeAttrHelper::get(const std::string& key, float def_val) {
+  for (int i = 0; i < node_.attribute_size(); i++) {
+    const ONNX_NAMESPACE::AttributeProto& attr = node_.attribute(i);
+    if (attr.name() == key) {
+      return attr.f();
+    }
+  }
+
+  return def_val;
+}
+
+int32_t NodeAttrHelper::get(const std::string& key, int32_t def_val) {
+  for (int i = 0; i < node_.attribute_size(); i++) {
+    const ONNX_NAMESPACE::AttributeProto& attr = node_.attribute(i);
+    if (attr.name() == key) {
+      int64_t val = attr.i();
+      return SafeInt<int32_t>(val);
+    }
+  }
+
+  return def_val;
+}
+
+string NodeAttrHelper::get(const std::string& key, string def_val) {
+  for (int i = 0; i < node_.attribute_size(); i++) {
+    const ONNX_NAMESPACE::AttributeProto& attr = node_.attribute(i);
+    if (attr.name() == key) {
+      return attr.s();
+    }
+  }
+
+  return def_val;
+}
+
+vector<int32_t> NodeAttrHelper::get(const std::string& key, vector<int32_t> def_val) {
+  if (!has_attr(key)) {
```

Comment:
You're doing a copy for the return value so def_val could be `const vector<int32_t>&`